sympy/combinatorics/fp_groups.py

```diff
@@ -1250,14 +1291,22 @@ def reidemeister_presentation(fp_grp, H, C=None):
     if not C:
         C = coset_enumeration_r(fp_grp, H)
     C.compress(); C.standardize()
-    define_schreier_generators(C)
+    define_schreier_generators(C, homomorphism=homomorphism)
     reidemeister_relators(C)
     gens, rels = C._schreier_generators, C._reidemeister_relators
     gens, rels = simplify_presentation(gens, rels, change_gens=True)
 
     C.schreier_generators = tuple(gens)
     C.reidemeister_relators = tuple(rels)
 
+    if homomorphism:
+        _gens = []
+        C.schreier_gen_elem = {}
+        for gen in gens:
+            C.schreier_gen_elem[gen] = C._schreier_gen_elem[str(gen)]
+            _gens.append(C.schreier_gen_elem[gen])
```

Comment:
Does `C.schreier_gen_elem` actually serve a purpose? It seems enough to just compute `_gens`. You could even do `_gens = [C._schreier_gen_elem[str(gen)] for gen in gens]` in just one line

Reply:
That could be removed. Thought I might be useful elsewhere by making it an attribute, but, I don't think it's necessary.